03_letters/jecofang/letter_frequency.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func letter(s string) map[rune]int {
+	m := map[rune]int{}
+	for _, v := range s {
+		m[v]++
+	}
+
+	return m
+}
+
+func sortLetters(m map[rune]int) []string {
+	keys := []int{}
+	for k := range m {
+		keys = append(keys, int(k))
+	}
+	sort.Ints(keys)
+
+	s := make([]string, len(m))
+	for i, k := range keys {
+		s[i] = fmt.Sprintf("%c:%d", k, m[rune(k)])
```

Comment:
It's clearer to write `s = append(s, …)` (and `s := make([]string, 0, len(m))` to initialise it).